src/core/src/op/convert_align_types.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/convert_align_types.hpp"
+
+#include "itt.hpp"
+#include "validation_util.hpp"
+
+namespace ov {
+namespace op {
+namespace {
+
+std::unordered_map<size_t, element::Type> bit_to_int{
```

Comment:
It can be replaced by helper function and using `switch` to return correct type, will be simpler then create map ( no dynamic memory used).